Replaced xhrpatch with a custom ajax function

tomivirkki opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-grid/306)
<!-- Reviewable:end -->

tehapo commented:
Rebased this branch to include latest changes from master.

tehapo commented:
Fixed the single linter error that was breaking the build.

Saulis commented:
I'm getting this on Chrome on the second Angular 2 example.

![Screen Shot 2016-01-11 at 13.57.08.png](https://files.reviewableusercontent.io/images/reviewable/1039468/1jQFAyfKZvXZlWjxRazy3wzz/Screen%20Shot%202016-01-11%20at%2013.57.08.png)

---

Reviewed 14 of 15 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kXqvuUnjkwfmgdXfd:951146773)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Doesn't relate to this change. Seems to occur in master also.

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kZl2CIk_i-pfr4pUm:1097181621)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
fyi: fixed the issue in master and rebased this branch

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kbCD_W11xtLZG3D2o:-1672088641)*
<!-- Sent from Reviewable.io -->

Saulis commented:
:shipit: 

---

Review status: 14 of 15 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kiUQy2QmQvCi4Tqwv:1148172450)*
<!-- Sent from Reviewable.io -->

manolo commented on vaadin-grid.html:
```diff
@@ -6,6 +6,19 @@
 <link rel='import' href='../polymer/polymer.html'>
 <script src="vaadin-grid.min.js"></script>
+<script>
+if (!window.vaadin) {
```
This is an issue using the component in vaadin framework, since window.vaadin already exists

tomivirkki replied:
How is it a problem? If the namespace already exists then sdm.js simply isn't imported (as it shouldn't). We don't run sdm with Vaadin Framework during Core Elements development.

manolo replied:
Well we could be interested sometime in fact I debugged vaadin-grid in SDM with other GWT apps running SDM but never pure vaadin ones.
Anyway, you are right this is a very corner case.
